n2survey/lime/survey.py

```diff
@@ -158,6 +159,76 @@ def read_responses(self, responses_file: str) -> None:
         self.responses = question_responses
         self.lime_system_info = system_info
 
+    def get_responses(self, question: str, labels: bool = False) -> pd.DataFrame:
+        """Get responses for a given question with or without labels
+
+        Args:
+            question (str): Question to get the responses for.
+            labels (bool, optional): If the response consists of labels or not (default False).
+
+        Raises:
+            ValueError: Unconsistent question types within question groups.
+            ValueError: Unknown question types.
+
+        Returns:
+            [pd.DataFrame]: The response for the selected question.
+        """
+        question_group = self.questions[self.questions.question_group == question]
+        question_types = np.unique(question_group.type)
+        question_responses = self.responses.loc[:, question_group.index]
+        if len(question_types) > 1:
+            raise ValueError(f'Question {question} has multiple types {question_types}.')
+        else:
+            question_type = question_types[0]
+
+        if question_type == 'single-choice':
+            # ASSUME: question_responses consists only one column
+            #         in result question_group also consists of only one item
+            # create a DataFrame because questions_response is only a Series
+            response = pd.DataFrame()
+            response[question] = question_responses[question_responses.columns[0]]
```

Comment:
1. I do not understand that correctly. I tried to understand by runing tests with question='A2' but 'A2' survey.response is not in the index. See line 179:
```
> question_responses = self.responses.loc[:, question_group.index]
...
KeyError: "['A2'] not in index"
> self.responses.columns[:10]
Index(['A00', 'A1', 'A2other', 'A3', 'A3other', 'A4', 'A4other', 'A5',
       'A5other', 'A6'],
      dtype='object')
```
Is that what you mean or is that another bug?

2. Due to I pick only one column it is only a series (I tested it) but if I replace `question_responses[question_responses.columns[0]]` by `question_responses[question_responses.columns]` it would work but I do not understand why there are two columns for single-choice answers? Also, is that always the case?

Reply:
3. You are right. My mistake when I thought about single-choice responses should contain one columns. I did remove the if statements for labels=False but we have to keep in mind that there will be no checks of the question types for labels=False.